0000-settlement-engine/0000-settlement-engine.md

```diff
@@ -0,0 +1,353 @@
+
+##### Request
+
+```http
+POST /accounts/:id/sendMessage HTTP/1.1
+Accept: application/octet-stream
+Content-Type: application/octet-stream
+```
+
+> _&lt;raw bytes of message&gt;_
+
+##### Response
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/octet-stream
+```
+
+> _&lt;raw bytes of response&gt;_
+
+### Idempotency
+
+_(The following section was adapted from the [Stripe API documentation](https://stripe.com/docs/api/idempotent_requests).)_
+
+Each API MUST support [idempotency](https://en.wikipedia.org/wiki/Idempotence) for safely retrying requests without accidentally performing the same operation twice. This is useful when an API call is disrupted in transit and you do not receive a response. For example, if a request to send a settlement does not respond due to a network connection error, you can retry the request with the same idempotency key to guarantee that no more than one charge is created.
+
+To perform an idempotent request, provide an additional `Idempotency-Key: <key>` header to the request.
+
+Idempotency works by saving the resulting status code and body of the first request made for any given idempotency key, regardless of whether it succeeded or failed.
+
+Subsequent requests with the same key return the same result, including `500` errors.
+
+An idempotency key is a unique value generated by the client which the server uses to recognize subsequent retries of the same request. How you create unique keys is up to you, but we suggest using V4 UUIDs, or another random string with enough entropy to avoid collisions.
+
+Keys expire after 24 hours, so a new request is generated if a key is reused outside of that time frame. The idempotency layer compares incoming parameters to those of the original request and errors unless they're the same to prevent accidental misuse.
+
+Results are only saved if an API endpoint started executing. If incoming parameters failed validation, or the request conflicted with another that was executing concurrently, no idempotent result is saved because no API endpoint began execution. It is safe to retry these requests.
+
+All `POST` requests accept idempotency keys. Sending idempotency keys in `GET` and `DELETE` requests has no effect and should be avoided, as these requests are idempotent by definition.
+
+## Usage in Interledger
+
+Peers using the [Interledger protocol](https://github.com/interledger/rfcs/tree/master/0001-interledger-architecture) clear and fulfill conditional IOUs which affect their mutual accounting relationship.
+
+Since Interledger merely provides a credit network, a peer's liabilities may accrue to the point at which they must reconcile them. Settlement engines provide a mechanism for peers in Interledger to settle their liabilities and continue transacting.
+
+An Interledger connector MAY operate the accounting system and transport system that interfaces with the settlement engine.
```

Comment:
It's probably worth a discussion around the value of abstraction in this RFC.

Currently, the spec seems to be trying to define a generic accounting/settlement system while stipulating that one of many use-cases could be an Interledger Connector. 

Instead, we should consider crafting this RFC from the perspective of: "this RFC is specific to Interledger" while at the same time leaving room, for example, for some of the components to be run outside of a Connector (maybe).